### ISSUE TITLE
Problems with Part Bulkloader

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#1242]
I have not yet been able to get this tool to work successfully, and need some guidance as to the source of the following error for the file of 43 records in the parts bulkloader:
Error Executing Database Query. 
[Macromedia][Oracle JDBC Driver][Oracle]ORA-06550: line 1, column 7: PLS-00306: wrong number or types of arguments in call to 'MOVEPARTTOCONTAINER' ORA-06550: line 1, column 7: PL/SQL: Statement ignored
This message has been logged. Please contact us with any information that might help us to resolve this problem.

I would like to be able to use the "use existing" function, but I am updating part condition and possibly part dispostion from what may already exist in Arctos. It would be very helpful if condition and/or dispostion could be set to stay the same as what already exists in Arctos when using "use existing". 
I think some of the issues with this tool not working may be if a part has been subsampled for loan?

**Comment by dustymc** [comment]
There is an update stuck in never-never land, waiting a server at TACC to be fixed. I'm pretty sure that this is just part of that.

I'm not sure what you're doing, but I don't think it's bulkloading. I'm happy to help, but I'm going to need a lot more information in order to do so.

Subsampling, for a loan or otherwise, establishes a relationship which disallows deletion of the donor. That's it.

**Comment by campmlc** [comment]
I need to load barcodes to existing unbarcoded parts. We have over 10,000,
so not something I want to do one at a time using bulk edit container. The
current 43 records were samples pulled for a loan and barcoded after
subsampling and then reinstalled. I'd like to add the barcodes to the parts
and then add the parts to the loan.

On Aug 17, 2017 9:30 AM, "dustymc" <notifications@github.com> wrote:

There is an update stuck in never-never land, waiting a server at TACC to
be fixed. I'm pretty sure that this is just part of that.

I'm not sure what you're doing, but I don't think it's bulkloading. I'm
happy to help, but I'm going to need a lot more information in order to do
so.

Subsampling, for a loan or otherwise, establishes a relationship which
disallows deletion of the donor. That's it.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ArctosDB/arctos/issues/1242#issuecomment-323108022>,
or mute
the thread

**Comment by dustymc** [comment]
I think that's best done via SQL, at least initially. The part bulkloader is focused on creating parts, and if it encounters any problems with the weird stuff it's willing to try it'll just create a new part. If you have any interest in being able to track loans back to specific parts (and I'm sure you do or you wouldn't be creating subsamples), the part bulkloader is not an appropriate tool for this - it'll almost inevitably create parts that will not be linked to your loans.

If in that process you ask me to do the same predictable thing a few times, I'll roll it into a new bulkloader. (That's how most all bulkloaders come to be.)

**Comment by campmlc** [comment]
Yes, I do want to track loans, and I need to be able to add these barcodes
to parts that have been loaned and subsampled previously, rather than
creating a new part. I don't want to duplicate identical parts.  It is
difficult to extract the conditon and dispositon information - I need a
tool that will make those optional unless I am updating or restricting the
parameters.

Can you use the data in the parts bulkloader for the SQL update? Let me
know what information you need. I would like to send this loan as soon as
possible, so I need to know whether I should wait on SQL entry or try to
manually enter these 43 barcodes one at a time.

Thank you!

**Comment by dustymc** [comment]
https://arctos.database.museum/guid/MSB:Mamm:228724 does not have a "liver (frozen)" part - what do you want me to do with it?

The rest seem to have one appropriate part so this should be straightforward.

Here's the plan; let me know if it sounds right:

1. find the part
2. move the part-container to the container corresponding to barcode in those data
2. update part condition to whatever's in the file
2. update part disposition to whatever's in the file

**Comment by campmlc** [comment]
Yes, that sounds good.

The MSB228724 part should be heart, kidney (frozen).

**Comment by campmlc** [comment]
Thank you!!
